### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by ankitt26**
### In this PR, I did-
- Create model classes for all entities as shown in the  [ERD](https://github.com/ankitt26/Ankye-blogs/blob/data-model/blog_app_erd.png) diagram.
- Set up associations in the User model
- Set up associations in the Post model
- Set up associations in the Comment model
- Set up associations in Like model

### Create custom methods -
- A  method `three_most_recent_posts` that returns the 3 most recent posts for a given user.

 - A method `update_posts_counter` that updates the posts counter for a user.
    
 - A method `five_most_recent_comments` which returns the 5 most recent comments for a given post.

 - A method `update_comments_counter` that updates the comments counter for a post.

 - A method `update_likes_counter` that updates the likes counter for a post.

### CODE REVIEW

File: app/models/like.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Like < ApplicationRecord
+  belongs_to :user, class_name: 'User'
```
Comment (cynthiainga) [inline app/models/like.rb]: - [ ] Good job so far, but the likes belong to `author` not `user`, kindly fix that 🙃

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  belongs_to :post, class_name: 'Post'
+
+  after_create :update_likes_counter
+
+  def update_likes_counter
+    post.increment!(:likes_counter)
+  end
```
Comment (cynthiainga) [inline app/models/like.rb]: - [OPTIONAL] Good job by one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html), but after using it you should make the `update_likes_counter` method private. Also, do the same for `update_comments_counter` and `update_posts_counter` methods 🙃 🙃

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @ankitt26, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] No linters errors ✔️ 
- [ ] Has `User`, `Post`, `Comment` and `Like` models inside `app/models/` directory ✔️ 
- [ ] Correct Gitflow ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇🏾 👇🏾 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/like.rb (hunk 1 above)
Comment (ankitt26) [reply app/models/like.rb]: Thanks for requesting changes, I am working on it 🙋🏿‍♂️.

File: app/models/like.rb (hunk 2 above)
Comment (ankitt26) [reply app/models/like.rb]: Thank you 😊.

**Review by laiifuu — APPROVED** [review]
# STATUS: APPROVED :heavy_check_mark: :tada: 

## Hi @ankitt26  :wave: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

friends-joey

## Highlights
- [x] Required changes made by the previous reviewer were implemented :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and :computer: Happy coding :computer: 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ankitt26** [comment]
Thanks for Approving the PR 😊.

**Pull request merged**